biochemist-python/chapters/Non-Linear Regression Part 2.ipynb

```diff
@@ -0,0 +1,794 @@
+    "<p class=\"overview-title\">Overview</p>\n",
+    "<p>Questions</p>\n",
+    "    <ul>\n",
+    "        <li> How can I perform statistically robust non-linear regression in a Jupyter notebook?</li>\n",
+    "    </ul>\n",
+    "<p>Objectives:</p>\n",
+    "    <ul>\n",
+    "        <li> Import data from a csv file\n",
+    "        <li> Create a function based on the equation to be fitted\n",
+    "        <li> Generate the best-fit curve with parameters and statistical descriptors\n",
+    "        <li> Create plots of the data\n",
+    "    </ul>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Data Fitting\n",
+    "In an earlier module, you used the `linregress` function from the scipy.stats library to perform linear regression. This is sufficient for a least squares linear fit to your data. Python contains tools that also enable you to fit data to an equation that you have defined, including two or more paramaters. So it is ideal for nonlinear curve fitting to the Michaelis-Menten equation.\n",
+    "\n",
+    "In this module, we will use some familiar tools: importing csv data into a pandas dataframe and plotting results using the matplotlib.pyplot library. We will also create a function based on the Michaelis-Menten equation. The exciting part is to learn to fit the data to the function using `curve_fit` from scipy.optimize.\n",
+    "\n",
+    "Here are the steps in our process:\n",
+    "\n",
+    "1. Import the libraries and functions that you need, including curve_fit from scipy.optimize.\n",
+    "1. Set up a pandas dataframe from the MM_data.csv file you saved from the previous module.\n",
+    "1. Create your independent and dependent variables for curve fitting.\n",
+    "1. Create a function for the equation you want to fit. The function should accept as inputs the independent variable(s) and all the parameters to be fit.\n",
+    "1. Use the function curve_fit to fit your data.\n",
+    "1. Extract the fit parameters from the output of curve_fit.\n",
+    "1. Use your function to calculate y values using your fit model to see how well your model fits the data.\n",
+    "1. Graph your original data and the fit equation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import the libraries and functions that you need, including curve_fit from scipy.optimize.\n",
+    "import os\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "from scipy.optimize import curve_fit"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Make sure that you are in the correct directory to do this work. The commands are slightly different depending on your operating system.\n",
+    "\n",
+    "Mac or Linux:  cd ~/Desktop/python-scripting-biochemistry  \n",
```

Comment:
As in the previous lesson, there is no reason to navigate in the Jupyter noteboook. Remove 'cd' then change the os.path.join command to start with the data directory.

Reply:
I removed this material.